### ISSUE TITLE
conda install misses pip dependencies in wheel MANIFEST file

### ISSUE THREAD

**Issue by mieczyslaw** [conda/conda#11983]
### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

_I added similar ticket to conda-build (https://github.com/conda/conda-build/issues/4595) but it is probably more related to `conda env create/conda install`._

conda-build docs mentioned that it is possible to install the package from wheel (built using python -m build), and build.sh should contain pip install *whl (where *whl is full wheel name).

The conda package is built, but when installed from the channel (yes, I know package installed with conda install package_name.tar.bz2 won't have dependencies installed - a weird feature btw), the pip dependencies as stated in *whl METADATA are not installed in the env (only conda packages are installed).

When `pip install` is called on the whl file in the shell in the already activated empty conda env, all pip dependencies are downloaded and installed.

It looks like the issue with `conda env create`/`conda install` that wheel METADATA entries (Requires-Dist) are not picked up for pure pip packages installation.

### Conda Info

```shell
conda version : 22.9.0
conda-build version : 3.21.9
python version : 3.9.13.final.0
```

**Comment by travishathaway** [comment]
Hi @mieczyslaw,

Could you help us by providing an example for how to recreate this and where exactly to look to find (or not find) the missing information.

This will help us get more background on the issue.

Thank you.

**Comment by mieczyslaw** [comment]
Hi @travishathaway ! I appreciate your help on that! I prepared the example and attached the simple project, so you don't have to do everything from scratch.

[mydummyproject.tar.gz](https://github.com/conda/conda/files/9843884/mydummyproject.tar.gz)

1. I created the sample project using a cookiecutter template:
`cookiecutter https://github.com/audreyfeldroy/cookiecutter-pypackage.git`

2. I removed unnecessary files and created `environment.yml`
```
channels:
    - conda-forge
    - nodefaults
dependencies:
    - python=3.9
    - pip>=22.0
    - pip:
          - -e .
```

3. I added `loguru` and `build` to `setup.py` - those are the pip dependencies. I will show later that they are missed when conda package is installed.

4. I called `conda env create --file environment.yml --prefix .env` and `conda list` - you'll see `loguru` installed.

5. I called `setup -m build` and you can find in `dist`: `mydummyproject-0.1.0-py2.py3-none-any.whl`. You can see inside: `mydummyproject-0.1.0.dist-info`, and inside it there is `METADATA` file with:
```
Requires-Dist: loguru
Requires-Dist: build
```
Those are the pip dependencies `conda install` will be missing.

6. Let's do another experiment - let's first create a fresh env (`environment2.yml`) and pip install the wheel (so no pip -e this time):
```
channels:
    - conda-forge
    - nodefaults
dependencies:
    - python=3.9
    - pip>=22.0
```

`conda env create --file environment2.yml --prefix .env2`

You won't see loguru yet when calling `conda list`, but it is fine at this stage. 

Call: `pip install dist/mydummyproject-0.1.0-py2.py3-none-any.whl` 

By calling `conda list` now you can see that `loguru` is installed.

7. Now let's build a conda package from wheel (building it out of pure source is not the best because there is no access to dependencies in private PyPi, and wheels will be build for all pip packages, so conda package becomes massive). I followed: https://docs.conda.io/projects/conda-build/en/latest/user-guide/wheel-files.html

Note: I had `conda-build` installed in my `(base)` env and I build the conda package when in `(base)` env.

Call: `conda build --no-anaconda-upload --no-include-recipe conda-recipe`

Examine created conda package `mydummyproject-0.1.0-py39_0.tar.bz2` (I copied it to `dist`) - `lib/python3.9/site-packages/mydummyproject-0.1.0.dist-info/METADATA` (the same file as in the `*whl` file so the wheel has been included into the conda package).

8. Create local channel (the same issue happens with the remote one) and place the built package in `conda_channel/linux-64`, then index (your `file://` path will be different later on depending where you keep your channel):
```
mkdir conda_channel
mkdir linux-64
# copy conda package from dist to linux-64
conda index conda_channel/
```

Add the channel to `.condarc`: `conda config --append channels file://home/mietek/development/conda_channel/`

9. Create a new env (`environment3.yml`) with the package from the channel
```
channels:
    - conda-forge
    - nodefaults
    - file://home/mietek/development/conda_channel/
dependencies:
    - python=3.9
    - pip>=22.0
    - mydummyproject=0.1.0
```
`conda env create --file environment3.yml --prefix .env3`

Activate the third env and you'll see that loguru is not installed, but `mydummyproject` is installed:

```
# packages in environment at /home/mietek/development/mydummyproject/.env3:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
bzip2                     1.0.8                h7f98852_4    conda-forge
ca-certificates           2022.9.24            ha878542_0    conda-forge
ld_impl_linux-64          2.39                 hc81fddc_0    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 12.2.0              h65d4601_18    conda-forge
libgomp                   12.2.0              h65d4601_18    conda-forge
libnsl                    2.0.0                h7f98852_0    conda-forge
libsqlite                 3.39.4               h753d276_0    conda-forge
libuuid                   2.32.1            h7f98852_1000    conda-forge
libzlib                   1.2.13               h166bdaf_4    conda-forge
mydummyproject            0.1.0                    py39_0    file://home/mietek/development/conda_channel
ncurses                   6.3                  h27087fc_1    conda-forge
openssl                   3.0.5                h166bdaf_2    conda-forge
pip                       22.3               pyhd8ed1ab_0    conda-forge
python                    3.9.13          h2660328_0_cpython    conda-forge
readline                  8.1.2                h0f457ee_0    conda-forge
setuptools                65.5.0             pyhd8ed1ab_0    conda-forge
sqlite                    3.39.4               h4ff8645_0    conda-forge
tk                        8.6.12               h27826a3_0    conda-forge
tzdata                    2022e                h191b570_0    conda-forge
wheel                     0.37.1             pyhd8ed1ab_0    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge
```

It means that pip dependencies in conda package are not parsed. The same happens when the package is installed from the server hosting conda channel. The solution would be to somehow call `pip install` on the included wheel.

**Comment by mieczyslaw** [comment]
@dholth Could you please kindly confirm whether there is any ongoing work on this ticket and whether the fix is likely to be available? I tried to look into `conda install` code myself, but I am not familiar enough to find a fix. Thanks in advance.

**Comment by mieczyslaw** [comment]
I found a temporary workaround by listing all dependencies from `require` section of `setup.cfg` in `post-link.sh` file to `pip install` them there. It seems to work, but against the recommendation that `post-link.sh` is not used for this purpose.

**Comment by travishathaway** [comment]
Hi @mieczyslaw,

My apologies for not looking at this in a while (we get quite a few issues on this project 😅).

I've added the https://github.com/conda/conda/labels/tag%3A%3Apip label to this issue as I believe it highlights current issues with our pip interoperability.  I cannot promise when we will be able to get to this issue, but I want to you let you know that the information you took the time to document will be very valuable to us going forward.

Please keep an eye on this issue for further updates.

**Comment by mieczyslaw** [comment]
Thanks for the update @travishathaway

**Comment by mieczyslaw** [comment]
I don't think this has been repaired.

**Comment by dholth** [comment]
Have you tried https://github.com/zuzukin/whl2conda

**Comment by mieczyslaw** [comment]
It is not a pure python dependency package, I am building a conda package with some conda only dependencies and then I need to add pip ones (private or pip only). The reason is that conda build does not allow to use private PyPi repos (it ignores pip.conf), and even for public ones conda install ignored '--constraints' parameter when it calls pip even though it was present in pip section of environment.yml (I created another ticket for that). Ideally when conda installs the package, it follows the content of MANIFEST file.

**Comment by dholth** [comment]
`MANIFEST` is the list of files in the wheel package, do you mean `METADATA`

**Comment by mieczyslaw** [comment]
yes, sorry I meant METADATA - basically when one calls conda install, those packages in metadata are not installed (so the workaround as mentioned above is to add them to post-link and let conda package to be a bit bigger as it will contain installed pip packages)
